contact_map/contact_count.py

```diff
@@ -83,16 +126,31 @@ class ContactCount(object):
         method to obtain the object associated with the number used in
         ``counter``; typically :meth:`mdtraj.Topology.residue` or
         :meth:`mdtraj.Topology.atom`.
-    n_x : int
-        number of objects in the x direction (used in plotting)
-    n_y : int
-        number of objects in the y direction (used in plotting)
+    n_x : int, tuple(start, end), optional
+        range of objects in the x direction (used in plotting)
+        Default tries to plot the least amount of symetric points.
+    n_y : int, tuple(start, end), optional
+        range of objects in the y direction (used in plotting)
+        Default tries to show the least amount of symetric points.
+    max_size : int, optional
+        maximum size of the count
+        (used to determine the shape of output matrices and dataframes)
     """
-    def __init__(self, counter, object_f, n_x, n_y):
+    def __init__(self, counter, object_f, n_x=None, n_y=None, max_size=None):
         self._counter = counter
         self._object_f = object_f
+        self.total_range = _get_total_counter_range(counter)
+        n_x, n_y = _sanitize_n_x_n_y(n_x, n_y, counter)
         self.n_x = n_x
         self.n_y = n_y
+        self.n_x_min, self.n_x_max = _int_or_range_to_tuple(n_x)
+        self.n_y_min, self.n_y_max = _int_or_range_to_tuple(n_y)
```

Comment:
It seems to me that there's enough (repeated) logic here that maybe it's worth refactoring into some small class `ContactPlotRange` (or any better name you can think of) so that here it could just be `self.n_x = ContactPlotRange(n_x, counter)`, and `self.n_x.min` and `self.n_x.max` would give the desired info (and perhaps `self.n_x.as_input`, although I'm not sure whether that is necessary to preserve here). You could also then create `@property` something like:

```python
@property
def range_length(self):
    return self.max - self.min
```

This gets used in, e.g., the DPI check, and would probably make that a little easier to read.